src/Symfony/Component/Validator/ConstraintValidatorFactory.php

```diff
@@ -42,7 +39,7 @@ public function getInstance(Constraint $constraint)
 
         if (!isset($this->validators[$className])) {
             $this->validators[$className] = 'validator.expression' === $className
```

Comment:
I think we should remove the service (well we might have to keep it for BC reasons but deprecate it) and remove the special handling of the `ExpressionValidator` here as it is no longer needed.